ran into a seemingly rare toctou issue here once:

2/6 Test #1: bugfixTests ......................***Failed    0.50 sec
Traceback (most recent call last):
  File "/builddir/exiv2-0.28.4/tests/runner.py", line 72, in <module>
    system_tests.configure_suite(conf_file)
  File "/builddir/exiv2-0.28.4/tests/system_tests.py", line 160, in configure_suite
    os.mkdir(abs_path)
FileExistsError: [Errno 17] File exists: '/builddir/exiv2-0.28.4/test/tmp'

--- a/tests/system_tests.py	2025-02-10 15:10:32.000000000 +0000
+++ b/tests/system_tests.py	2025-02-12 13:50:29.660557367 +0000
@@ -157,7 +157,10 @@
                 os.path.join(_parameters["suite_root"], rel_path)
             )
             if key == "tmp_path" and not os.path.isdir(abs_path):
-                os.mkdir(abs_path)
+                try:
+                    os.mkdir(abs_path)
+                except FileExistsError:
+                    pass
             if key == "data_path" and not os.path.exists(abs_path):
                 raise ValueError(
                     "Path replacement for {short}: {abspath} does not exist"
